src/backend/tests/it/get_users.rs

```diff
@@ -0,0 +1,186 @@
+use std::time::{Duration, UNIX_EPOCH};
+
+use crate::utils::{
+    mock::{CALLER, ISSUER_CANISTER_ID, VC_HOLDER, VP_JWT},
+    pocketic::{query_call, setup, update_call},
+};
+use candid::Principal;
+use ic_verifiable_credentials::issuer_api::CredentialSpec;
+use pocket_ic::PocketIc;
+use shared::types::user_profile::{
+    AddUserCredentialError, AddUserCredentialRequest, GetUsersRequest, GetUsersResponse, OisyUser,
+    UserProfile,
+};
+
+fn create_users(pic_setup: &(PocketIc, Principal), start: u8, end: u8) {
+    for i in start..=end {
+        let caller = Principal::self_authenticating(i.to_string());
+        let response = update_call::<UserProfile>(pic_setup, caller, "create_user_profile", ());
+        assert!(response.is_ok());
+    }
+}
+
+#[test]
+fn test_get_users_returns_users() {
+    let pic_setup = setup();
+
+    let users_count = 5;
+    create_users(&pic_setup, 1, users_count);
+
+    let caller = Principal::from_text(CALLER).unwrap();
+
+    let arg = GetUsersRequest {
+        matches_max_length: None,
+        updated_after_timestamp: None,
+    };
+    let get_users_response = query_call::<GetUsersResponse>(&pic_setup, caller, "get_users", arg);
+
+    assert_eq!(
```

Comment:
in this test and other below, in addition to the length can we also compare the returned users themselves?

See for example `assert_custom_tokens_eq`

Reply:
Good idea, I added it to this test, but I don't see the value in doing it everywhere.

Creating the expected users adds boilerplate and complexity and I don't see how that will improve the other tests.

Reply:
> I don't see how that will improve the other tests.

Iv'e got the feeling that it can proove itself to be useful when we will start adding more fields in the profile. So I see it as an "investement" for the future.

Reply:
Good point!

Like this it made sense also that create_users had some delay creating each user so that each has a different timestmamp as well.